### ISSUE TITLE
Packaging request / Help building: evolution-ews

### ISSUE THREAD

**Issue by sgillespie** [NixOS/nixpkgs#67281]
**Project description**
Microsoft Exchange server support for Evolution

**Metadata**

* homepage URL: https://wiki.gnome.org/Apps/Evolution
* source URL: https://gitlab.gnome.org/GNOME/evolution-ews.git
* license: lgpl
* platforms: linux

I've begun the process here: https://github.com/sgillespie/nixpkgs/commit/05c592e84e9a8ebe5376569e41b1e59590a70673. Unfortunately this doesn't build because it's trying to update evolution-data-server, and I'm not smart enough to get past it.

```
-- Installing: /nix/store/r1jiqcmdjmzai3zm8hkvw8xsmx8jl32a-evolution-data-server-3.32.4/lib/evolution-data-server/addressbook-backends/libebookbackendews.so
CMake Error at src/addressbook/cmake_install.cmake:55 (file):
  file INSTALL cannot copy file
  "/build/evolution-ews-3.32.2/build/src/addressbook/libebookbackendews.so"
  to
  "/nix/store/r1jiqcmdjmzai3zm8hkvw8xsmx8jl32a-evolution-data-server-3.32.4/lib/evolution-data-server/addressbook-backends/libebookbackendews.so".
Call Stack (most recent call first):
  src/cmake_install.cmake:42 (include)
  cmake_install.cmake:55 (include)


FAILED: CMakeFiles/install.util
cd /build/evolution-ews-3.32.2/build && /nix/store/rlp98mmbvp008720p25k73w2khzs9rzn-cmake-3.14.5/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
builder for '/nix/store/dhnhdfbhvzqzwpfiw62nyizsp77ab224-evolution-ews-3.32.2.drv' failed with exit code 1
error: build of '/nix/store/dhnhdfbhvzqzwpfiw62nyizsp77ab224-evolution-ews-3.32.2.drv' failed
```

**Comment by bjornfor** [comment]
Cross reference: https://github.com/NixOS/nixpkgs/issues/12847.

**Comment by jtojnar** [comment]
It uses pkg-config to find the directory. Unfortunately CMake does not support [proper variable handling](https://gitlab.kitware.com/cmake/cmake/issues/19632), so you will need to override the variable manually using environment variable.

For example, if you see

```cmake
pkg_check_variable(ebook_backenddir libedata-book-1.2 backenddir)
```

you can set the override environment variable as suggested in `pkg-config(1)` section ENVIRONMENT VARIABLES:

```nix
PKG_CONFIG_LIBEDATA_BOOK_1_2_BACKENDDIR = "${placeholder "out"}/lib/evolution-data-server/addressbook-backends";
```

**Comment by jtojnar** [comment]
Also note that you will probably need to create a non-trivial patch for `evolution-data-server` for it to be able to find it. I can probably take a jab at it after we update GNOME to 3.34.

**Comment by aanderse** [comment]
@jtojnar any chance you have capacity to look at this now that 3.34 is in unstable?

**Comment by jtojnar** [comment]
Moving it up in my GNOME to-do list just below dconf database module but I will not be able to get to it until February.

**Comment by aanderse** [comment]
Great, thanks!

**Comment by fabianhauser** [comment]
This is still valid and important to me. Current state as far as I see: @jtojnar planned to take a look at patching the `evolution-data-server`.

**Comment by jtojnar** [comment]
This is being handled in https://github.com/NixOS/nixpkgs/pull/91504 and #92775

**Comment by das-g** [comment]
#91504 has been merged on 2020-07-09, but I guess #92775 will also be needed to make use of it to connect with Evolution to Exchange servers, won't it?

Or should this issue here (#67281) be closed as a duplicate of #12847?

**Comment by jerith666** [comment]
This was done in #103135.
